static/js/search_suggestion.js

```diff
@@ -583,6 +618,9 @@ exports.get_suggestions = function (query) {
     suggestions = get_operator_suggestions(last);
     attach_suggestions(result, base, suggestions);
 
+    suggestions = get_has_filter_suggestions(last, base_operators);
```

Comment:
Seperate functions for is and has filters were created instead of adding the suggestion for has into `special_filter_suggestions` to take care of the different priorities of the suggestions for both the filters. `has` suggestions were previously obtained from `get_operator_suggestions`, therefore placing it directly below that. Also it makes sense for `has` filter to have a low priority than is as filters like `private messages` and `starred messages` maybe used a lot more than has operators.